CAIPs/caip-285.md

```diff
@@ -0,0 +1,126 @@
+---
+caip: 285
+title: JSON-RPC Method for Revoking Session Authorizations
+author: [Alex Donesky] (@adonesky1)
+discussions-to: TBD
+status: Draft
+type: Standard
+created: 2024-07-12
+requires: 25, 217
+---
+
+## Simple Summary
+
+CAIP-285 introduces the `wallet_revokeSession` method for fully revoking the authorizations and properties of an active [CAIP-25][] session.
+
+## Abstract
+
+This proposal aims to enhance session management for [CAIP-25][] initiated sessions by defining a new JSON-RPC method for revoking sessions. This method provides an explicit protocol for revoking sessions with or without `sessionId`s.
+
+## Motivation
+
+The motivation behind this proposal is to enhance the flexibility of [CAIP-25][] initated sessions by enabling the revocation of session authorizations without `sessionId`s, which don't map well to extension-based wallet's dapp connections and could add unnecessary constraints and burdens to existing flows. The proposed method provides an intuitive way to revoke authorizations of an active session, simplifying the management of session lifecycles.
+
+## Specification
+
+### Definition
+
+The `wallet_revokeSession` method revokes the entire active session. If a `sessionId` is provided, it revokes that specific session; otherwise, it revokes the single active session between the wallet and the caller
```

Comment:
```suggestion
The `wallet_revokeSession` method revokes the entire active session.
If a `sessionId` is provided, it revokes that specific session only; if none is provide and there is only one session to revoke, it gets revokes and a success result is returned; otherwise, an appropriate error message is sent.
```

Reply:
Seems right except I would say:
```
if none is provided and there is an active session without a `sessionId`, it gets revokes and a success result is returned;
```
Since `if none is provided and there is only one session to revoke` seems to me to make ambiguous whether there can be multiple active sessions without `sessionIds`

Reply:
updated with some hybrid of our language here: https://github.com/ChainAgnostic/CAIPs/pull/285/commits/405c5a03d7d0386c4f3d87bcf8db8de81af6257f